client.go

```diff
@@ -304,9 +316,60 @@ func (client *client) GetOffset(topic string, partitionID int32, time int64) (in
 	return offset, err
 }
 
+func (client *client) Coordinator(consumerGroup string) (*Broker, error) {
+	client.lock.RLock()
+	coordinator := client.coordinators[consumerGroup]
+	client.lock.RUnlock()
+
+	if coordinator == nil {
+		var err error
+		err = client.RefreshCoordinator(consumerGroup)
+		if err != nil {
+			return nil, err
+		}
+
+		client.lock.RLock()
+		coordinator = client.coordinators[consumerGroup]
+		client.lock.RUnlock()
+	}
+
+	_ = coordinator.Open(client.conf)
+	return coordinator, nil
+}
+
+func (client *client) RefreshCoordinator(consumerGroup string) error {
+	coordinator, err := client.getCoordinator(consumerGroup, client.conf.Metadata.Retry.Max)
+	if err != nil {
+		return err
+	}
+
+	client.lock.Lock()
+	client.coordinators[consumerGroup] = client.registerBroker(coordinator)
```

Comment:
- Primary reason: the `getCoordinator` method has enough to worry about. It's easier to understand it and refactor it later if we limit the scope.
- Nobody can access `getCoordinator` anyway because it's not exported. Only within sarama itself this is possible, so `anybody` is quite limited in scope. Not sure about the need to protect against ourselves; it feels defensive given that we're all responsible adults here. :P 
- Even if you do end up calling `getCoordinator` without updating the cache, it doesn't cause any issues besides doing potentially unnecessary requests to the cluster.
- It's easier to test stuff in isolation when functions do not have side effects.